Automattic/wp-calypso
# Trail Map: separate feature list for structure-only variant

Related to #90477

## Proposed Changes

* Creates a separate feature list for the structure-only variant. This is needed since we want to customize the order of features in the list and also omit certain features from appearing for a plan. Eg: "Support via email" should not be visible for the Creator plan.
* Also removes `includePreviousPlanFeatures` from everywhere since it is no longer required.
* TODO: There are still some features that are not quite aligned:
  * Free domain
 
  <img width="1345" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/be4ed2fb-cf88-453e-add6-a80abbc948be">

  * Growth and monetization tools
  <img width="1354" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/813a87fd-100d-42e9-8b4c-54323351e529">

#### Full screenshot
![localhost_64949_iframe html_args= id=featuresgrid--trail-map-structure viewMode=story](https://github.com/Automattic/wp-calypso/assets/5436027/5e3c56f9-03d0-4c1d-9ce9-2d21945586b9)

## Testing Instructions



* Go to `/start/plans` and confirm that the features grid and comparison grid are unchanged from production.
* Run `yarn workspace @automattic/plans-grid-next storybook`.
* Check the Trail Map Structure story for the Feature Grid.
* Confirm that the structure variant contains all features from the control variant.

## Pre-merge Checklist
